### ISSUE TITLE
code should not require three argument multiply

### ISSUE THREAD

**Issue by JeffreySarnoff** [JuliaDiff/TaylorSeries.jl#62]
in Taylor1 there is this line

``` julia
        coefhomog += (kcoef-i) * ac[kcoef-i+1] * coeffs[i+1]
```

which looks for *(::Int64, :::Float64, ::Float64)
It would be more portable to supply the parens

**Comment by JeffreySarnoff** [comment]
hmm --- looks like vectors -- never mind

**Comment by JeffreySarnoff** [comment]
reopening this because something is awkward -- deserves a look

**Comment by lbenet** [comment]
Thanks for taking a look on this!

This kind of expressions appear often when functions of polynomials are computed. For concreteness, I'll assume that you are looking into `TaylorSeries.expHomogCoef`, and I will use the corresponding formula given in the [documentation](http://www.juliadiff.org/TaylorSeries.jl/latest/api/#TaylorSeries.expHomogCoef). Note that this formula provides a *recursive* relation to compute the coefficients of `c=exp(a)`, with `a` a polynomial in one variable.

`Taylor1{T}` objects are stored as a vector of `T`-coefficients, and the order (maximum degree) of the polynomial. So, indeed, `ac` and `coeffs` seem to be related with the  coefficients of `a` (the polynomial) and the already computed coefficients of `c = exp(a)`, in the example I am considering.

> reopening this because something is awkward -- deserves a look 

Can you be more specific in what you find as awkward? (I agree that something is awkward  deserves a look).

**Comment by JeffreySarnoff** [comment]
```julia
using TaylorSeries
using ArbFloats
setprecision(ArbFloat,28);

t1=Taylor1([ArbFloat(2), ArbFloat(3)])
# 2.00000000 + 3.00000000 t + 𝒪(t²)
evaluate(t1,ArbFloat(0.5))
# 3.50000000
#   that is good
quit()
```
```julia
using ArbFloats
using TaylorSeries
setprecision(ArbFloat,28);

 t1=Taylor1([ArbFloat(2), ArbFloat(3)])
# 2.00000000 + 3.00000000 t + 𝒪(t²)
evaluate(t1,ArbFloat(0.5))
ERROR: * not defined for ArbFloats.ArbFloat{28}    
 in evaluate(::TaylorSeries.Taylor1{ArbFloats.ArbFloat{28}}, ::ArbFloats.ArbFloat{28}) at      
  /home/jas/.julia/v0.5/TaylorSeries/src/Taylor1.jl:875   
 in eval(::Module, ::Any) at ./boot.jl:234   
 in macro expansion at ./REPL.jl:92 [inlined]  
 in (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at ./event.jl:46  

# that is awkward
```
the only difference is the order of the `using` statements

**Comment by lbenet** [comment]
Closing since it is (somehow) fixed in #95.
